src/energyplus/Test/CoilSystemCoolingWaterHeatExchangerAssisted_GTest.cpp

```diff
@@ -0,0 +1,295 @@
+
+  // They must be connected to a PlantLoop too
+  PlantLoop chw_p(m);
+  chw_p.addDemandBranchForComponent(coil.coolingCoil());
+
+  ForwardTranslator ft;
+  Workspace w = ft.translateModel(m);
+
+  WorkspaceObjectVector idfUnitarys(w.getObjectsByType(IddObjectType::AirLoopHVAC_UnitarySystem));
+  ASSERT_EQ(1u, idfUnitarys.size());
+  WorkspaceObject idfUnitary(idfUnitarys[0]);
+
+  WorkspaceObjectVector idfCoilSystems(w.getObjectsByType(IddObjectType::CoilSystem_Cooling_Water_HeatExchangerAssisted));
+  ASSERT_EQ(1u, idfCoilSystems.size());
+  WorkspaceObject idfCoilSystem(idfCoilSystems[0]);
+
+  WorkspaceObjectVector idfHXs(w.getObjectsByType(IddObjectType::HeatExchanger_AirToAir_SensibleAndLatent));
+  ASSERT_EQ(1u, idfHXs.size());
+  WorkspaceObject idfHX(idfHXs[0]);
+
+  WorkspaceObjectVector idfCoils(w.getObjectsByType(IddObjectType::Coil_Cooling_Water));
+  ASSERT_EQ(1u, idfCoils.size());
+  WorkspaceObject idfCoil(idfCoils[0]);
+
+  // No CoilSystem:Cooling:Water wrapper needed, it's inside a unitary
+  EXPECT_EQ(0, w.getObjectsByType(IddObjectType::CoilSystem_Cooling_Water).size());
+
+  // Check that the Unitary ends up with the CoilSystemCoolingWaterHeatExchangerAssisted
+  EXPECT_EQ("CoilSystem:Cooling:Water:HeatExchangerAssisted", idfUnitary.getString(AirLoopHVAC_UnitarySystemFields::CoolingCoilObjectType).get());
+  EXPECT_EQ(idfCoilSystem.nameString(), idfUnitary.getString(AirLoopHVAC_UnitarySystemFields::CoolingCoilName).get());
+
+  // Since the Unitary only has a Cooling Coil (no fan, not HC / Suppl HC), the nodes should match
+  EXPECT_EQ(idfUnitary.getString(AirLoopHVAC_UnitarySystemFields::AirInletNodeName).get(),
+            idfHX.getString(HeatExchanger_AirToAir_SensibleAndLatentFields::SupplyAirInletNodeName).get());
+
+  EXPECT_EQ(idfUnitary.getString(AirLoopHVAC_UnitarySystemFields::AirOutletNodeName).get(),
+            idfHX.getString(HeatExchanger_AirToAir_SensibleAndLatentFields::ExhaustAirOutletNodeName).get());
+
+  EXPECT_EQ("My CoilSystemCoolingWaterHeatExchangerAssisted",
+            idfCoilSystem.getString(CoilSystem_Cooling_Water_HeatExchangerAssistedFields::Name).get());
+  EXPECT_EQ("HeatExchanger:AirToAir:SensibleAndLatent",
+            idfCoilSystem.getString(CoilSystem_Cooling_Water_HeatExchangerAssistedFields::HeatExchangerObjectType).get());
+  EXPECT_EQ(coil.heatExchanger().nameString(),
+            idfCoilSystem.getString(CoilSystem_Cooling_Water_HeatExchangerAssistedFields::HeatExchangerName).get());
+  EXPECT_EQ("Coil:Cooling:Water", idfCoilSystem.getString(CoilSystem_Cooling_Water_HeatExchangerAssistedFields::CoolingCoilObjectType).get());
+  EXPECT_EQ(coil.coolingCoil().nameString(), idfCoilSystem.getString(CoilSystem_Cooling_Water_HeatExchangerAssistedFields::CoolingCoilName).get());
+
+  // Check Controller:WaterCoil created by coil.addToNode
+  EXPECT_EQ(1, w.getObjectsByType(IddObjectType::Controller_WaterCoil).size());
```

Comment:
I think this is right, but what is actually creating/setting this controller? It doesn't seem like `coilCoolingWater.addToNode()` is being called...

Reply:
I think the answer here is `PlantLoop_Impl::addDemandBranchForComponent()`.